### DISCUSSION TITLE
SEO Optimization with Apollo

### DISCUSSION THREAD

**Discussion by mshahbazm** [vercel/next.js#11318] [Help] [answered]
I am following [ with-apollo](https://github.com/zeit/next.js/tree/canary/examples/with-apollo) example to make my web application, however the meta tags seem to be not rendered for the bots, i.e. if I debug my dynamic url say http://abc.com/item/[id]/[name]
at https://developers.facebook.com/tools/debug/ this always gets the default meta tags that I have set inside _app, even though I am updating meta tags with props (received from getInitialProps) inside component using next/head. 
I have tried this using getServerSideProps as well.
Is this something doable with nextjs?
Please help, I have googled alot and tried almost any example that seemed to be solution, but no luck so far. 
I am using Next 9.3 and apollo/client 3.

**Comment by mshahbazm** [comment, marked as answer]
I have figured it out. It is possible with next.js, It was silly mistake on my side. I was not calling API request inside the getInitialProps.

**Comment by HaNdTriX** [comment]
Just in case somebody is looking at this discussion.

The default behaviour of the `withApollo` HOC is:

```js
export default withApollo({ ssr: false })(PageComponent)
```

You can opt into full SSR (meaning apollo queries are fetched on the server) by enabling `ssr`:

```diff
- export default withApollo({ ssr: false })(PageComponent)
+ export default withApollo({ ssr: true })(PageComponent) 
```

Please note that this technique is then using `AppTree`-Traversal wich is pretty slow compared to other data fetching methods. Thats why we disabled it per default.

https://github.com/zeit/next.js/blob/canary/examples/with-apollo/pages/index.js
